pkg/otlp/internal/serializerexporter/consumer.go

```diff
@@ -58,8 +59,21 @@ func (c *serializerConsumer) ConsumeTimeSeries(ctx context.Context, name string,
 	)
 }
 
+// addTelemetryMetric to know if an Agent is using OTLP metrics.
+func (c *serializerConsumer) addTelemetryMetric(hostname string) {
+	c.series = append(c.series, &metrics.Serie{
+		Name:           "datadog.agent.otlp.metrics",
+		Points:         []metrics.Point{{Value: 1, Ts: float64(time.Now().Unix())}},
+		Tags:           []string{},
+		Host:           hostname,
+		MType:          metrics.APIGaugeType,
+		SourceTypeName: "System",
+	})
+}
+
 // flush all metrics and sketches in consumer.
-func (c *serializerConsumer) flush(s serializer.MetricSerializer) error {
+func (c *serializerConsumer) flush(hostname string, s serializer.MetricSerializer) error {
+	c.addTelemetryMetric(hostname)
```

Comment:
Nit: Any reason to have this inside the `flush` method instead of calling it from `ConsumeMetrics`?

Reply:
Moved to `ConsumeMetrics`